### ISSUE TITLE
CA1822: false-positive when using conditional compilation

### ISSUE THREAD

**Issue by zhaparoff** [dotnet/roslyn-analyzers#3107]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

Example: [CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822?view=vs-2019)

#### Repro steps
```CSharp
private bool MethodA(int value)
{
    // Non-static, using instance members
}

private bool MethodB(int value) // <-- CA1822
{
#if DEBUG
    MethodA(value);
#endif
    return value;
}
```

#### Expected behavior
Rule should not trigger, since the code will be broken, when DEBUG flag will be set.

#### Actual behavior
Rule triggered

**Comment by mavasani** [comment]
This is a duplicate of https://github.com/dotnet/roslyn/issues/31820#issuecomment-447602659. It is not possible for the analysis to account for all possible syntax trees for the entire matrix of preprocessor directives. The code should be rewritten so that parameter value is only defined in debug build.
